src/engine/sparqlExpressions/SparqlExpressionValueGetters.cpp

```diff
@@ -171,3 +171,35 @@ IntDoubleStr ToNumericValueGetter::operator()(
     [[maybe_unused]] const EvaluationContext* context) const {
   return std::string(asStringViewUnsafe(s.getContent()));
 }
+
+// ____________________________________________________________________________
+LiteralOrString makeDatatypeValueGetter::operator()(
+    ValueId id, const EvaluationContext* context) const {
+  // 1. false: don't remove the brackets (easier to make an explicit literal
+  // later on)  2. false: we also want to consider other "typed" types w.r.t. to
+  // DATATYPE()
+  auto optionalStringAndType =
+      ExportQueryExecutionTrees::idToStringAndType<false, false, std::identity>(
+          context->_qec.getIndex(), id, context->_localVocab);
+  if (!optionalStringAndType.has_value()) {
+    return std::monostate{};
```

Comment:
Unfortunately this doesn't work here as expected, 
as the `idToStringAndType` function does rather strange things for the datatypes which only somewhat seem to work inside the exporting, where the typing is lost anyway (It somewhat works for the literal content of a string, that's why we currently use it for those expressions.

So the correct way to do this is to implement the functionality from scratch (a function that switches on the type of the `ValueId`.
(See for the implementation of the `idToStringAndType` for what you have to do for `VocabIndex` and `LocalVocabIndex`.)